src/MIDebugEngine/Engine.Impl/DebuggedThread.cs

```diff
@@ -44,7 +44,7 @@ internal class ThreadCache
         private List<DebuggedThread> _deadThreads;
         private List<DebuggedThread> _newThreads;
         private Dictionary<string, List<int>> _threadGroups;
-        private static uint s_targetId = uint.MaxValue;
+        private static uint s_targetId = 0; // Thread ids should be start at 0 in the default scenario
```

Comment:
Can you talk about why you changed this part? Unfortunately there are no comments to say for sure, but I **think** the idea of starting with `uint.MaxValue` was to select something that was highly unlikely to clash with actual thread ids in the case that the thread id couldn't be parsed.

This code was added in commit [12632f3280da94787c009c8a97accbe797ac6994](https://github.com/microsoft/MIEngine/commit/12632f3280da94787c009c8a97accbe797ac6994), but the author didn't add any more details, and I am sure the author no longer remembers.